### DISCUSSION TITLE
How do I use overrideWithValue the same way overrideWithProvider was used? (v 1.x)

### DISCUSSION THREAD

**Discussion by ketanchoyal** [rrousselGit/riverpod#788] [Q&A] [answered]
I'm using riverpod v1.x for my apps and in the latest dev release overrideWithProvider is removed,

I have a provider in the client package which I have override to the following provider:

```
final ClientSNPToOverride = StateNotifierProvider<Client, ClientProps>((ref) {
return Client.init(
    url: buildConfig.baseUrl,
    reader: ref.read,
    secretBox: HiveStore.secret,
    cacheBox: HiveStore.cache,
    logoutFunction: () {
      ref.read(authServiceP).signOut();
    },
  );
});
```

I have to pass the Reader function to the init constructor which I don't know how do I pass when I use overrideWithValue

I've used overrideWithProvider very extensively so any suggestions will be appreciated

**Comment by rrousselGit** [comment, marked as answer]
Rather than overriding the provider, you can override one of its dependency

So if needed, you could split your provider to allow overriding, such as:

```dart
final secretProvider = Provider<String>(...)

StateNotifierProvider<Client, ClientProps>((ref) {
  return Client.init(
   ...
    secretBox: ref.watch(secret),
  );
});

...
ProviderScope(
  overrides: [secretProvider.overrideWithValue(HiveStore.secret)],
)
```

**Reply by ketanchoyal** [reply]
Yes, this should work perfectly, thanks for the quick reply
